packages/nx/src/tasks-runner/cache.ts

```diff
@@ -29,13 +30,37 @@ export type CachedResult = {
 };
 export type TaskWithCachedResult = { task: Task; cachedResult: CachedResult };
 
+// This function is called once during tasks runner initialization. It checks if the db cache is enabled and logs a warning if it is not.
 export function dbCacheEnabled(nxJson: NxJsonConfiguration = readNxJson()) {
-  return (
-    !IS_WASM &&
-    process.env.NX_DISABLE_DB !== 'true' &&
-    nxJson.useLegacyCache !== true &&
-    process.env.NX_DB_CACHE !== 'false'
-  );
+  // If the user has explicitly disabled the db cache, we can warn...
+  if (
+    nxJson.useLegacyCache ||
+    process.env.NX_DISABLE_DB === 'true' ||
+    process.env.NX_DB_CACHE === 'false'
+  ) {
+    // TODO (@AgentEnder): Insert correct link after @isaacplmann has written the documentation.
+    logger.warn(
+      `Nx is using the filesystem cache instead of the db cache. There are some limitations and trade-offs to consider when using the filesystem cache. Read more at https://nx.dev/{insert-link-here}.`
```

Comment:
I think we can call it the legacy cache and mention that it will be removed in Nx 21. Instruct them to start looking into moving to the new db based cache.